src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildWorkspaceUpgradeContext.cs

```diff
@@ -94,7 +97,7 @@ public IProject GetOrAddProject(FileInfo path)
                 return cached;
             }
 
-            var project = new MSBuildProject(this, _componentIdentifier, path, _logger);
+            var project = new MSBuildProject(this, _componentIdentifier, _restorer, path, _logger);
```

Comment:
this is odd, it looks like a hidden dependency.

Is the thought process here that the MSBuildWorkspace and MSBuildProject types are always used together in pairs?

It could make it easier to test if this was a constructor dependency.

Reply:
yeah, that would be nice